### DISCUSSION TITLE
Source Code temporary Pull Request comments

### DISCUSSION THREAD

**Discussion by paulo-rico** [community/community#127208] [Pull Requests]
### Select Topic Area

General

### Body

In brief, here is my experience / problem / solution (happened many times)

1. I am contributing to a repository quite an extensive code change with new algorithms replacing old ones, i.e., many deletions / additions.
2. I wish to heavily comment my code so the project owners can understand the algorithms easier and quicker (this is a very popular repository with an active community forum the owners are constantly monitoring / resolving issues, so the less time they require to understand new requests, the better).
3. If merged, the owner doesn't want the repository littered with these explanatory comments.

Would it be possible to have the following -

1. Some sort of comment character combination for GitHub Pull Request explanatory notes only
2. Ability to view how new file would look without those comments
3. When merged into repository, these GitHub Pull Request comments would be ignored.

What does the community think?

**Comment by 7Cdrs** [comment]
You can propose using a special comment marker like //#PRONLY: for temporary PR-specific comments. These comments will help reviewers understand your changes but will be stripped out before merging. A script or GitHub Action can be used to remove these comments automatically. This approach keeps the review process clear and the final codebase clean. What does the community think?
